VultisigApp/VultisigApp/Chains/Ton.swift

```diff
@@ -14,53 +14,128 @@ enum TonHelper {
+        let transfer: TheOpenNetworkTransfer
         
-        let transfer = TheOpenNetworkTransfer.with {
-            $0.dest = toAddress.description
-            // When using attachAllContractBalance mode, amount is ignored, so set to 0 for clarity
-            $0.amount = sendMaxAmount ? 0 : (UInt64(keysignPayload.toAmount.description) ?? 0)
-            $0.mode = sendMode
+        // Check if this is a jetton transfer
+        if !jettonAddress.isEmpty {
             
-            if let memo = keysignPayload.memo  {
-                $0.comment = memo
+            // Build jetton transfer
+            transfer = try buildJettonTransfer(keysignPayload: keysignPayload, jettonAddress: jettonAddress, isActiveDestination: isActiveDestination)
+        } else {
+            
+            // Build native TON transfer
+            guard let toAddress = AnyAddress(string: keysignPayload.toAddress, coin: .ton) else {
+                throw HelperError.runtimeError("fail to get to address")
+            }
+            
+            let baseMode = TheOpenNetworkSendMode.ignoreActionPhaseErrors.rawValue
+            var sendMode = UInt32(TheOpenNetworkSendMode.payFeesSeparately.rawValue | baseMode)
+            if sendMaxAmount {
+                sendMode = UInt32(TheOpenNetworkSendMode.attachAllContractBalance.rawValue | baseMode)
+            }
+            
+            transfer = TheOpenNetworkTransfer.with {
+                $0.dest = toAddress.description
+                $0.amount = sendMaxAmount ? 0 : (UInt64(keysignPayload.toAmount.description) ?? 0)
+                $0.mode = sendMode
+                
+                if let memo = keysignPayload.memo {
+                    $0.comment = memo
+                }
+                $0.bounceable = bounceable
             }
-            $0.bounceable = bounceable
         }
         
+        let sequenceNumberUInt32 = UInt32(sequenceNumber.description) ?? 0
+        let expireAtUInt32 = UInt32(expireAt.description) ?? 0
+        
         let input = TheOpenNetworkSigningInput.with {
             $0.messages = [transfer]
-            $0.sequenceNumber = UInt32(sequenceNumber.description) ?? 0
-            $0.expireAt = UInt32(expireAt.description) ?? 0
+            $0.sequenceNumber = sequenceNumberUInt32
+            $0.expireAt = expireAtUInt32
             $0.walletVersion = TheOpenNetworkWalletVersion.walletV4R2
             $0.publicKey = pubKeyData
         }
         
-        return try input.serializedData()
+        let serializedData = try input.serializedData()
+        
+        return serializedData
+    }
+    
+    static func buildJettonTransfer(keysignPayload: KeysignPayload, jettonAddress: String, isActiveDestination: Bool) throws -> TheOpenNetworkTransfer {
+        
+        // Convert destination to bounceable, as jettons addresses are always EQ
+        let destinationAddress = try convertToUserFriendly(address: keysignPayload.toAddress, bounceable: true, testOnly: false)
```

Comment:
why `bounceable` is hard coded to true? it should respect the one in blockchainspecific